docs/runtimes/application.ipynb

```diff
@@ -82,16 +82,306 @@
+    "    return {\"status_code\": requests.get(\"https://docs.mlrun.org/\").status_code}\n",
+    "\n",
+    "\n",
+    "@app.route('/internal')\n",
+    "def internal():\n",
+    "    # Test access to the serving function with MLRun.\n",
+    "    project_name = \"serving-project\"\n",
+    "    project = mlrun.get_or_create_project(project_name)\n",
+    "    function = project.get_function('add-ten-serving')\n",
+    "    response = function.invoke('/v2/models/add-ten-model/infer', {\"inputs\": [20]})\n",
+    "    output = response['outputs']['outputs']\n",
+    "    return {\"result\": output}\n",
+    "\n",
+    "\n",
+    "@app.route('/external')\n",
+    "def external():\n",
+    "    # Test access to the serving function without MLRun (externally).\n",
+    "    project_name = \"serving-project\"\n",
+    "    project = mlrun.get_or_create_project(project_name)\n",
+    "    function = project.get_function('add-ten-serving')\n",
+    "    url = f\"https://{function.status.external_invocation_urls[0]}\"\n",
+    "    response = requests.post(url, json={\"inputs\": [50]}).json()\n",
+    "    output = response['outputs']['outputs']\n",
+    "    return {'result': output}\n",
+    "\n",
+    "\n",
+    "@app.route('/secret')\n",
+    "def secret():\n",
+    "    # Every application has access to project secrets.\n",
+    "    # This is just an example; do not return real secrets!\n",
+    "    project_secret = os.environ.get('secret-example')\n",
+    "    return {'secret': project_secret}\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "Let's save this code block as Python files and zip it into a .tar.gz file:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "!tar -czvf archive.tar.gz flask_app_example.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "Now, we are going to deploy this flask application:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "import mlrun\n",
+    "project = mlrun.get_or_create_project(\"app-demo-flask\")\n",
+    "# Create a demo secret for testing\n",
+    "project.set_secrets(secrets={\"secret-example\":\"project_secret_example\"})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
    "metadata": {
     "collapsed": false
-   }
+   },
+   "outputs": [],
+   "source": [
+    "# Specify source to be loaded on build time or run time\n",
+    "# The image or the requirements should include mlrun, flask and gunicorn.\n",
+    "application = project.set_function(name=\"flask_app\", kind=\"application\", image=\"your-image\", requirements_file=\"/your/path/requirements.txt\")"
```

Comment:
You don't actually have to set an image, because in this case you build an application image.
If you do set it, and also set commands and args, the image will be used as a base image for the build.

Reply:
image arg removed, tnx